### PR TITLE
Move globalThis polyfill to compatibility, add more compatibility

### LINKED ISSUE #16355: Here is a working solution: iOS 9.3.5 (iPad2) and Homeassistant
### Checklist

- [X] I have updated to the latest available Home Assistant version.
- [X] I have cleared the cache of my browser.
- [X] I have tried a different browser to see if it is related to my browser.

### Describe the issue you are experiencing

actual homeassistant does not work with old iPads (ios 9.3.5)

### Describe the behavior you expected

a working solution using safari on old iPads

### Steps to reproduce the issue

after login on the browser nothin happens because of browser/javascript engine to old

### What version of Home Assistant Core has the issue?

docker 2023.1

### What was the last working version of Home Assistant Core?

docker 2020.1

### In which browser are you experiencing the issue with?

safari 9

### Which operating system are you using to run this browser?

iOS 9.3.5

### State of relevant entities

```txt
(TODO: add to wiki !)

Many had the same troubles like me:
Old iPads mounted to the wall are good enough as a touchscreen BUT
Lovelace UI is incompatible with iOS 9.3.5 the latest version for old iPads.

Behavior: you can connect to the homeassistant but after login you see "initializing" and wait forever.

Reason: Lovelace UI uses Polymere Javascript framework which needs newer browsers like Safari 11

Updating to the last 2 supported Browsers (Dolphin (tested), iCab (not tested €€€)) do not work because Dolphin uses iOS builtin Webkit Browserengine.

What works for some people: install homeassistant app on an iphone of a family member and then on the old ipad in appstore under family sharing (de: Familienfreigabe) you can download the latest compatible version of that app.

Solution: 2 steps

1) the last version of homeassistant docker container that kind of works with workaround is: 2021.12
Next version 2022.1 does not, even with workaround.  2021.12 has working knx integration

2) login token does still not work with 2021.12 so you need to switch to an alternatin authentication method, like trusted networks

---------------------------------------------------------

Here is what you need to do (in this order !!!)

1)
sudo docker pull homeassistant/home-assistant:2021.12

2)
sudo docker run -d \
 --restart unless-stopped \
 --name=homeassistant \
 -h homeassistant \
 -e PUID=1001 \
 -e PGID=1001 \
 -e TZ=CET \
 --net=host \
 -v homeassistant:/config \
    homeassistant/home-assistant:2021.12

3) connect on a PC and configure a user (doesn't work with trusted networks)
http://homeassistant.local:8123

4) enable trusted networks in configuration.yaml (insert first 5 rows end change to your net adress)
sudo nano /var/lib/docker/volumes/homeassistant/_data/configuration.yaml

----------------------------------------
homeassistant:
  auth_providers:
    - type: trusted_networks
      trusted_networks:
        - 192.168.8.0/24

# Configure a default setup of Home Assistant (frontend, api, etc)
default_config:

# Text to speech
tts:
  - platform: google_translate

group: !include groups.yaml
automation: !include automations.yaml
script: !include scripts.yaml
scene: !include scenes.yaml
knx: !include knx.yaml      # if you need knx integration

-------------------------------------------------------------

5) (optional) add your knx configuration (no need for configuring IP interface or IP router)
this is my knx.yaml:
--------------------------------------------------------------

---
#knx:

  light:
  - name: Deckenlicht_Glastüre
    address: 0/0/4
  - name: Bar_Licht1
    address: 0/0/10
  - name: Wandlicht_Glastüre_Stiege
    address: 0/0/11

  cover:
  - name: Raffstore_Küche_Hoch_Tief
    move_long_address: 1/0/12
  - name: Raffstore_Küche_Stop
    move_long_address: 1/0/13
  - name: Raffstore Küche Pos1
    position_address: 1/0/27
  - name: Raffstore Küche Pos2
    position_address: 1/0/28

  switch:
  - name: Sofa_Stehlampe_steckdose_rechts
    address: 0/0/3
  - name: Alles_Aus1
    address: 0/5/1
  - name: Stiege_Steckdose_Christbaum
    address: 2/0/11

  sensor:
  - name: Strommessung_all_abfragen
    state_address: 5/0/0
    type: "active_energy"
    state_class: "total_increasing"
 
----------------------------------------------------------

6)  restart homeassistant
(ether via web browser on PC or via:
docker restart homeassistant 
)

7)  open safari on your iPad2 (iOS 9.3.5) and open the url:
http://homeassistant.local:8123
select your user and have fun!
```

### Problem-relevant frontend configuration

```yaml
iPad2 iOS 9.3.5
```

### Javascript errors shown in your browser console/inspector

_No response_

### Additional information

This is the solution so many people are looking for.
please add here to issues as well as to wiki to make it findable in the future

### PR DESCRIPTION
**Pull request by bramkragten**
## Proposed change

All modern browsers that we target today have support for `globalThis`

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [X] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes https://github.com/home-assistant/frontend/issues/16355
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/panels/config/integrations/ha-integration-card.ts [hunk 1]
```diff
@@ -1003,7 +1003,7 @@ export class HaIntegrationCard extends LitElement {
         @media (min-width: 563px) {
           ha-card.group {
             position: relative;
-            min-height: 200px;
+            min-height: 195px;
```
Comment (steverep) [inline src/panels/config/integrations/ha-integration-card.ts:1006]: Did you accidentally branch this off another PR?

File: src/resources/compatibility.ts [hunk 2]
```diff
@@ -1,6 +1,7 @@
 // Caution before editing - For latest builds, this module is replaced with emptiness and thus not imported (see build-scripts/bundle.js)
 import "core-js";
 import "regenerator-runtime/runtime";
+import "globalthis-polyfill";
```
Comment (steverep) [inline src/resources/compatibility.ts:4]: I don't think we need another package - `core-js` should already be polyfilling this?

https://github.com/zloirock/core-js#ecmascript-globalthis

Comment (bramkragten) [reply src/resources/compatibility.ts:4]: I'll test

**Comment by bramkragten** [comment]
There is still some race condition, where it sometime still errors on missing `globalThis`, after a refresh it does work...

**Review by steverep — DISMISSED** [review]

**Comment by bramkragten** [comment]
> What module is throwing the error for you?

Lit 🙃

**Comment by bramkragten** [comment]
This should make the frontend work on iOS 9 now 🤞

**Comment by steverep** [comment]
I'm guessing the race is just because core and app are loaded in parallel and core has the polyfills.  I've got a fix for that planned soon.

**Comment by bramkragten** [comment]
For ES5 we should probably not loaded in parallel, for latest we should

**Review by steverep — APPROVED** [review]

**Comment by bramkragten** [comment]
Just bought an iPad 3 to actually test it, as browserstack won't let me 😢

**Comment by steverep** [comment]
Wait, spoke too soon... they are loaded with `<script defer>`, so theoretically they should go in order, right?

My thought was to use Babel's "usage" option instead of "entry", so it injects where the individual polyfills are actually needed.  Required a couple tweaks to get it compiled but I needed to test.  I throw up a branch.

**Comment by bramkragten** [comment]
Yes, scripts with the defer attribute will execute in the order in which they appear in the document.

**Comment by felamachado** [comment]
Im running HAOS, please write a manual of hownto do this.
I have an ios 9 3.5 iPad mini 1

**Pull request merged**